lib/fishjam/room.ex

```diff
@@ -401,14 +412,34 @@ defmodule Fishjam.Room do
   @impl true
   def handle_info(%EndpointMessage{endpoint_id: to, message: {:media_event, data}}, state) do
     with {:ok, peer} <- State.fetch_peer(state, to),
-         socket_pid when is_pid(socket_pid) <- Map.get(peer, :socket_pid) do
-      send(socket_pid, {:media_event, data})
+         socket_pid when is_pid(socket_pid) <- Map.get(peer, :socket_pid),
+         node_name when not is_nil(node_name) <- Map.get(peer, :node_name),
+         {:ok, rpc_result} <-
+           RPCClient.call(node_name, PeerSocketHandler, :send_media_event, [socket_pid, data]),
+         :ok <- rpc_result do
+      :ok
     else
+      :peer_socket_not_exists ->
+        Logger.warning(
+          "Received Media Event from RTC Engine to peer #{inspect(to)} which peer socket doesn't exist"
+        )
+
+        {:ok, peer} = State.fetch_peer(state, to)
+        disconnect_peer(peer.socket_pid, self(), :peer_socket_not_exists)
+
       nil ->
         Logger.warning(
           "Received Media Event from RTC Engine to peer #{inspect(to)} without established signaling connection"
         )
 
+      :error_rpc ->
+        Logger.warning(
+          "Received Media Event from RTC Engine to peer #{inspect(to)} which was connected to different node and rpc to that node failed."
+        )
+
+        {:ok, peer} = State.fetch_peer(state, to)
+        disconnect_peer(peer.socket_pid, self(), :peer_socket_not_exists)
```

Comment:
```suggestion
        disconnect_peer(peer.socket_pid, self(), :error_rpc)
```

or something like that